src/watcher.ts

```diff
@@ -1,39 +1,61 @@
 import chokidar from 'chokidar';
 import { exec, ChildProcess } from 'child_process';
+import type Serverless from 'serverless';
 
-export type Config = {
+type ConfigItem = {
   path: string | string[];
-  command: string;
-}[];
+  command?: string;
+  hooks?: string | string[];
+};
+
+export type Config = ConfigItem[];
 
 export interface Watcher {
   start: () => void;
   stop: () => Promise<void>;
 }
 
-export const makeWatcher = (config: Config): Watcher => {
+export const makeWatcher = (config: Config, serverless: Serverless): Watcher => {
   const internalWatchers: chokidar.FSWatcher[] = [];
   const processes: { [pid: number]: ChildProcess } = {};
 
+  const runCommand = (command: string, eventType: string, eventPath: string) => new Promise<void>((resolve) => {
```

Comment:
nit: We could probably take `runCommand` and `runHook` out of `makeWatcher` (adding serverless as an arg to runHook). Think this might make the code slightly simpler and discourage future changes that tangle up these methods with makeWatcher args.

Reply:
`runCommand` also requires the process mapping, and with that I'm running into another ESLint error :thinking: 

```javascript
      delete processes[call.pid];
// Assignment to property of function parameter 'processes' (no-param-reassign)
```
I can of course make this change the maping to be immutable and return new copies from `runCommand` if we want. 

But tbh I'm kinda starting to feel like [twitter.com/markdalgleish/status/1153595512248926208](https://twitter.com/markdalgleish/status/1153595512248926208) :sweat_smile:

Reply:
Haha. Forgot about this one. In that case let's keep them in this function.